app/models/user.rb

```diff
@@ -0,0 +1,14 @@
+class User < ActiveRecord::Base
+	has_many :todos
+
+	def authenticate(password)
+		hash = BCrypt::Engine.hash_secret(password, self.password_salt)
+		self.password_hash == hash	
+	end
+
+	def encrypt_password(password)
+		self.password_salt = BCrypt::Engine.generate_salt
+		self.password_hash = BCrypt::Engine.hash_secret(password, self.password_salt)
+	end
+
+end
```

Comment:
Thanks for the great effort by yeah, I agree with KL. Leverage the use of gems for other stuff so you can devote your time to others that need to be done.